convey/conveymetric/metric.go

```diff
@@ -0,0 +1,50 @@
+package conveymetric
+
+import (
+	"github.com/Comcast/webpa-common/convey"
+	"github.com/go-kit/kit/metrics"
+)
+
+// UnknownLabel is a constant for when key/tag can not be found in the C JSON
+const UnknownLabel = "unknown"
+
+// MetricClosure will be returned after update of struct, this should be used to update the struct, aka decrement the count
+type MetricClosure func()
+
+// CMetric provides
+type CMetric interface {
+	// Update takes the convey JSON to update internal struct, and return a closure to update the struct again, or an
+	// error
+	//
+	// Note: MetricClosure should only be called once.
+	Update(data convey.C) (MetricClosure, error)
+}
+
+// NewConveyMetric produces a CMetric where gauge is the internal structure to update, tag is the key in the C JSON
+// to update the gauge, and metricName is the `key` for the gauge cardinality.
+//
+// Note: The Gauge must have the metricName as one of the constant labels
+func NewConveyMetric(gauge metrics.Gauge, tag string, metricName string) CMetric {
+	return &cMetric{
+		tag:   tag,
+		metricName:  metricName,
+		gauge: gauge,
+	}
+}
+
+// cMetric is the internal CMetric implementation
+type cMetric struct {
+	tag   string
+	metricName  string
```

Comment:
This is called a "label" in the metrics APIs, both go-kit and Prometheus.  The metric name is actually the name of the gauge itself (unknown to this code).